proposals/p2760.md

```diff
@@ -0,0 +1,794 @@
+-   One syntax for constraints in either `where` clauses or `require`
+    declarations.
+-   Want to open up the syntax to expressing more general constraints.
+-   Syntax for a `require` declaration in an interface or named constraint:
+
+    > `require` _where-constraint-argument_ `;`
+
+-   Syntax for an `extend` declaration in an interface or named constraint:
+
+    > `extend` _facet-type-expression_ `;`
+
+## Future work: mixins
+
+Mixins have not been defined in a proposal so far. However, part of the process
+of resolving
+[issue #995](https://github.com/carbon-language/carbon-lang/issues/995) was
+deciding on a syntax for including a mixin in a class. This was done in order to
+make sure that class declarations that included names from another entity were
+treated consistently, for example always starting with the `extend` keyword.
+
+```
+// Mixin declarations and definitions are
+// outside the scope of this proposal.
+mixin M1;
+mixin M2;
+
+class C {
+  // Mixing in mixin M1
+  extend m: M1;
+
+  // Mixing in mixin M2. This member is not named.
+  // Initialized using `M2.MakeDefault()`.
+  extend _: M2 = M2.MakeDefault();
+
+  // Alternative to the above `M2` that uses a
+  // private name instead of no name:
+  extend private m2: M2;
+}
+```
+
+The declaration that a class uses a mixin is called a "mix" declaration. The
+syntax of a mix declaration is:
+
+<!-- prettier-ignore -->
+> `extend` [`private`|`protected`] (`_`|_id_) `:` _mixin-expression_ [`=`
+> _initializer-expression_] `;`
+
+The _id_ part of the mix declaration defines the name assigned to that mixin
+subobject. This name is may be used to access members of the mixin and to
+initialize the mixin in a constructor for the class. The optional `private` or
+`protected` access specifier controls the access to this name.
+
+With this proposal, base class declarations appear in the body of the class
+definition, like data members, so decision of whether mixins are more like base
+classes or data members of issue
+[#1000: Mixins: base classes or data members?](https://github.com/carbon-language/carbon-lang/issues/1000)
+is less significant. Like base classes, the mix declaration syntax begins with
+`extend`. Like data members, a class may have multiple mix declarations and they
+may be intermixed with field declarations. The layout of the memory of an object
+reflects the order of the declarations in the class body, defining the order of
+the mixin and field subobjects.
+
+## Rationale
+
+The main reason for the new syntax is consistency and simplification:
+
+-   The use of the `extend` keyword is the consistent way to mark what other
+    entities are consulted during name lookup.
+-   The `class` declaration is simplified by moving more into the definition
+    body.
+-   Making adapters a kind of class removes a kind of top-level declaration,
+    simplification, and matches how base classes are declared, a consistency.
```

Comment:
```suggestion
    a simplification, and matches how base classes are declared, a consistency.
```
(would have been slightly easier for me to parse, but feel free to ignore this comment).